pkg/api/v1/types.go

```diff
@@ -1249,6 +1249,11 @@ type PodSpec struct {
 
 // PodSecurityContext holds pod-level security attributes and common container settings.
 type PodSecurityContext struct {
+	// SupplementalGroups can be used to specify a list of
+	// additional groups which the main container process will run
+	// as. This will be applied to all containers in the pod in
+	// addition to the primary group of the container.
+	SupplementalGroups []int `json:"supplementalGroups,omitempty"`
```

Comment:
I'm a big fan of strong types.  It is one of the things on which Go is
pretty close to right, IMO

On Fri, Oct 16, 2015 at 6:07 AM, Paul Weil <notifications@github.com> wrote:

> In pkg/api/v1/types.go
> <https://github.com/kubernetes/kubernetes/pull/14991#discussion_r42239708>
> :
>
> > @@ -1249,6 +1249,11 @@ type PodSpec struct {
> >
> >  // PodSecurityContext holds pod-level security attributes and common container settings.
> >  type PodSecurityContext struct {
> > +	// SupplementalGroups can be used to specify a list of
> > +	// additional groups which the main container process will run
> > +	// as. This will be applied to all containers in the pod in
> > +	// addition to the primary group of the container.
> > +	SupplementalGroups []int `json:"supplementalGroups,omitempty"`
>
> +1 for the type. /cc @pmorie <https://github.com/pmorie> for fsgroup.
> @swagiaal <https://github.com/swagiaal> - when this change is made I'll
> need to rebase my OpenShift policy work for the new types.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/14991/files#r42239708>.
>